### ISSUE TITLE
False positive no-member for TypeAlias __origin__

### ISSUE THREAD

**Issue by jamesbraza** [PyCQA/pylint#8455]
### Bug description

# pylint: disable=missing-module-docstring

from typing import TypeAlias

DictAlias: TypeAlias = dict[int, float]
assert DictAlias.__origin__ is dict

### Configuration

```ini
n/a
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
a.py:6:7: E1101: Class 'dict' has no '__origin__' member (no-member)
```

### Expected behavior

No `no-member` to be emitted here

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.10 (main, Feb 19 2023, 17:57:18) [Clang 14.0.0 (clang-1400.0.29.102)]
```

### OS / Environment

macOS Monterey version 12.6

**Comment by nickdrozd** [comment]
https://docs.python.org/3/library/stdtypes.html?highlight=origin#genericalias.__origin__
